src/etc/inc/plugins.inc.d/kea.inc

```diff
@@ -45,6 +45,68 @@ function kea_services()
     return $services;
 }
 
+function kea_run()
+{
+    return [
+        'static_mapping' => 'kea_staticmap',
+    ];
+}
+
+function kea_staticmap($proto = null, $valid_addresses = true, $domain_fallback = 'not.found', $ifconfig_details = null)
+{
+    $staticmap = [];
+
+    if ($proto) {
+        $proto = (int)$proto;
+    }
+
+    if (($proto ?? 4) == 4) {
+        $keav4 = new \OPNsense\Kea\KeaDhcpv4();
+
+        if (!empty((string)$keav4->general->enabled)) {
+            $reservations = $keav4->reservations;
+            $reservation = $reservations->reservation;
+
+            foreach ($reservations->reservation->iterateItems() as $reservation) {
+                $ip_address = (string)$reservation->ip_address;
+                if (empty($ip_address) && $valid_addresses) {
+                    continue;
+                }
+                $hostname = (string)$reservation->hostname;
+                if (empty($hostname)) {
+                    $hostname = null;
+                }
+                $description = (string)$reservation->description;
+                if (empty($description)) {
+                    $description = null;
+                }
+
+                $subnet_node = $keav4->getNodeByReference("subnets.subnet4.{$reservation->subnet}");
+
+                $domain = $domain_fallback;
```

Comment:
we should maybe considering removing domain_fallback or at least not implementing it for kea.. it was only a workaround for unbound when it wasn't capable of setting the right domain

Reply:
So it should not use the system domain as fallback? 

https://github.com/opnsense/core/blob/24.7.a/src/etc/inc/plugins.inc.d/unbound.inc#L586